src/indexable.cr

```diff
@@ -412,24 +412,80 @@ module Indexable(T)
     indexes.map { |index| self[index] }
   end
 
+  # Calls the given block once for each index in `self`, passing `self`
+  # and `other` index element as parameter.
+  #
+  # ```
+  # a = [1, 2, 3]
+  # b = ["a", "b", "c"]
+  #
+  # a.zip(b)  { |x,y| puts "[ #{x} -- #{y} ]" }
+  # ```
+  #
+  # produces:
+  #
+  # ```text
+  # [ 1 -- a ]
+  # [ 2 -- b ]
+  # [ 3 -- c ]
+  #```
   def zip(other : Indexable)
```

Comment:
For documentation purposes, I would prefer to have the block argument in the method signature even if it is not used as a captured block: This makes it clear that this method definition is expected to be called with a block and it shows the block types:
```crystal
def zip(other : Indexable(U), &block : T, U ->) forall U
```

Reply:
👍 Completely agree with your point, it makes extremely more readable!

It also brings me a question:

The Indexable module is used as a mixin in other classes/structs what adds to them the `zip` methods behavior in addition to others. Currently only `Array#zip(?)` has tests covering it. Would make sense to move the specs to `src/std/indexable_spec.cr` instead of  `src/std/array_spec.cr`?